### DISCUSSION TITLE
.= operator, for calling a method or property and assigning the result back to the left-hand variable

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#4948] [Language Ideas] [answered]
For instance:

```
string info = "this is a secret string";
info.=Replace("secret", "redacted");
```

would be shorthand for:

```
string info = "this is a secret string";
info = info.Replace("secret", "redacted");
```

And this could also be used with properties, but of course the return type of the method or property would have to be assignable to the type of the variable on the left-hand side.

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/3051
